subiquity/models/filesystem.py

```diff
@@ -927,7 +928,10 @@ def available(self):
 
     @property
     def _number(self):
-        return self.device._partitions.index(self) + 1
+        if self.preserve:
+            return self.number
+        else:
+            return self.device._partitions.index(self) + 1
```

Comment:
This works because we don't currently allow adding additional partitions on a disk which is preserved right?

Reply:
Yes, either all partitions are old or all are new. We will need to think much harder about partition numbers when we support editing partition tables.